packages/astro/test/fixtures/with-endpoint-routes/src/pages/images/hex.ts

```diff
@@ -1,9 +1,6 @@
-import { readFileSync } from "node:fs";
+import { readFile } from 'node:fs/promises';
 
-export async function GET({ params, request }) {
-  const buffer = readFileSync(new URL('../../astro.png', import.meta.url));
-  return {
-    body: buffer.toString('hex'),
-    encoding: 'hex',
-  };
+export async function GET() {
+	const buffer = await readFile(new URL('../../astro.png', import.meta.url));
+	return new Response(buffer.buffer);
```

Comment:
TIL readFile with promises return a correctly encoded buffer compared to the sync version. I'm not sure why.